Assignment-1-Calculator.c

```diff
@@ -0,0 +1,130 @@
+#include<stdio.h>
+
+
+/* To check the error
+error - 1 for  invalid expression 
+error -2 for division by zero error */
+int error = 0;
+
+
+// The function is to check the priority of the operator according to the DMAS
+int priority_op(char operator){
+    if(operator == '*' || operator == '/')return 1;
+    if(operator == '+' || operator == '-')return 0;
+    return -1;
+}
+
+
+/* The fuction to  calculate the input */
+int calculate(char input[]){
+
+    int numbers[1000];  // array to store numbers or operands
+    char operators[1000]; // array to store operators
+
+    int num_idx =0;  
+    int op_idx =-1;
+
+    int i=0;
+    while(input[i] !='\0'){
+
+        if(input[i] == ' '){  //skip blank space
+            i++;
+            continue;
+        }
+
+        // push to numbers array if it is only number 
+        if(input[i] >= '0' && input[i]<='9'){
+            int temp_num=0;
+            while(input[i]>='0' && input[i] <='9'){
+                temp_num = temp_num*10 + (input[i]-'0');
+                i++;
+            }
+            numbers[num_idx] = temp_num;
+            num_idx++;
+        }
+        else if( input[i] == '*' || input[i] == '/' || input[i] == '+' || input[i] == '-'){

```

Comment:
rather than hardcoding the value */+- at all places, you can add as macro and use the same at all other places.

take care of hardcoded values.